### DISCUSSION TITLE
Cargo で依存クレートを更新する

### DISCUSSION THREAD

**Discussion by hayashi3017** [hayashi3017/sandbox#146] [General]
[cargo update - The Cargo Book](https://doc.rust-lang.org/cargo/commands/cargo-update.html)

**Comment by hayashi3017** [comment]
[killercup/cargo-edit: A utility for managing cargo dependencies from the command line.](https://github.com/killercup/cargo-edit)

**Comment by hayashi3017** [comment]
[kbknapp/cargo-outdated: A cargo subcommand for displaying when Rust dependencies are out of date](https://github.com/kbknapp/cargo-outdated)

**Comment by gejustin** [comment]
## Work Frank's Take

I set up the current structure today with Gary, so I have context on what we were optimizing for. Here's where I land:

### What's right

**`roadmap:fep` + `roadmap:aip` dual-labeling.** This is the real insight. Context engineering, the migration framework, validation tooling — they serve both roadmaps. Forcing them into one bucket is wrong. Dual labels solve it cleanly.

**`type:` labels for hierarchy.** GH issues are flat. Labels are the cheapest way to add structure without fighting the tool. Goal → Project → Task is the right depth.

**Goal issues as containers with sub-issues.** We already wired this up today via the sub-issues API. #26 (Goal 3) has #12, #13, #15, #16 as sub-issues. It renders hierarchy and progress in the UI. Works.

### Where I disagree

**One board with views vs. separate boards.** I'd keep 3 boards. Reasons:
- Different teams, different cadences. FEP is heads-down on DS migration. AIP is still finalizing goals.
- Side quests are explicitly *not* roadmap work — mixing them in adds noise.
- Boards are free. Filtered views on one board get cluttered when you have 20+ issues.
- You can still create a cross-cutting view on each board if needed.

**Milestones as separate issues.** Overkill. A task list header inside a project issue is enough. Don't create issues that exist only to contain other issues — that's ceremony for ceremony's sake.

### Answers to open questions

1. **type:milestone worth it?** No. Checklist header inside a project issue. Keep it lightweight.
2. **Side quests same board?** No. Different purpose, different cadence. The barn wood doesn't belong on the roadmap board.
3. **type:spike or type:research?** We already have a `research` label. That's sufficient.
4. **Nesting depth?** Goal → Project → Task. Three levels. If you need a fourth level, your project is too big — split it.

### Proposed action

1. Add `roadmap:fep`, `roadmap:aip` labels
2. Add `type:goal`, `type:project`, `type:task` labels
3. Relabel existing issues accordingly
4. Keep 3 boards, add cross-cutting items to both roadmap boards
5. Leave wiki as strategy/thesis layer (already working)

The labels and dual-membership idea is the real win here. The rest is refinement.
